server/app/auth/ProfileFactory.java

```diff
@@ -28,19 +29,22 @@ public class ProfileFactory {
   private final Provider<VersionRepository> versionRepositoryProvider;
   private final Provider<ApiKeyService> apiKeyService;
   private final Provider<UserRepository> userRepositoryProvider;
+  private final FeatureFlags featureFlags;
 
   @Inject
   public ProfileFactory(
       DatabaseExecutionContext dbContext,
       HttpExecutionContext httpContext,
       Provider<VersionRepository> versionRepositoryProvider,
       Provider<ApiKeyService> apiKeyService,
-      Provider<UserRepository> userRepositoryProvider) {
+      Provider<UserRepository> userRepositoryProvider,
+      FeatureFlags featureFlags) {
     this.dbContext = Preconditions.checkNotNull(dbContext);
     this.httpContext = Preconditions.checkNotNull(httpContext);
     this.versionRepositoryProvider = Preconditions.checkNotNull(versionRepositoryProvider);
     this.apiKeyService = Preconditions.checkNotNull(apiKeyService);
     this.userRepositoryProvider = Preconditions.checkNotNull(userRepositoryProvider);
+    this.featureFlags = featureFlags;
```

Comment:
Please wrap this in a call to `Preconditions.checkNotNull(`, similar to the other arguments.